tests/validation/tests/v3_api/test_workload.py

```diff
@@ -995,83 +982,70 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
         user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
         p_client = get_project_client_for_token(project, user_token)
         ns = rbac_get_namespace()
-        workload = p_client.create_workload(name=name,
-                                            containers=con,
-                                            namespaceId=ns.id,
-                                            daemonSetConfig={})
+        workload = create_workload_by_type(p_client, name, con, ns, config)
         wait_for_wl_to_active(p_client, workload)
         with pytest.raises(ApiError) as e:
             user_token = rbac_get_user_token_by_role(role)
             p_client = get_project_client_for_token(project, user_token)
             con = [{"name": "test1", "image": os.environ.get('RANCHER_TEST_IMAGE',
                                                          "nginx")}]
             p_client.update(workload, containers=con)
+            wait_for_pods_in_workload(p_client, workload)
+            validate_workload(p_client, workload, "deployment", ns.name)
```

Comment:
the type is not only deployment.

Reply:
Changed to `validate_workload(p_client, workload, config, ns.name)`
This will validate per workload type `["daemonSet", "statefulSet", "cronJob", "job"]`